phac-nml/irida-next
# Project Samples: Select all

## What does this PR do and why?
_Describe in detail what your merge request does and why._

Adds the ability to select/deselect all samples in a project.

1. Added a checkbox to the first header that, when checked, selects all samples in the project 
2. Added the selection count and total samples to the table footer.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/11295750/74bf2a9b-e697-4c79-b471-ba2e32dc8f6f)

![image](https://github.com/phac-nml/irida-next/assets/11295750/e1f2bcc3-1049-42fb-b990-b97bfebc93f2)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

- Go to a project that has samples (e.g. [Outbreak 2023](http://localhost:3000/yersinia/yersinia-pseudotuberculosis/outbreak-2023/-/samples))
- Click the select all checkbox, all samples should be selected
- Click the select all checkbox again, all samples should be de-selected
- Check to make sure the table footer is updated with the correct number of samples.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
